synapse/federation/sender/__init__.py

```diff
@@ -328,42 +340,57 @@ async def handle_event(event: EventBase) -> None:
                         )
                     }
 
+                    destinations.discard(self.server_name)
+
                     if send_on_behalf_of is not None:
                         # If we are sending the event on behalf of another server
                         # then it already has the event and there is no reason to
                         # send the event to it.
                         destinations.discard(send_on_behalf_of)
 
-                    logger.debug("Sending %s to %r", event, destinations)
-
                     if destinations:
-                        await self._send_pdu(event, destinations)
-
                         now = self.clock.time_msec()
                         ts = await self.store.get_received_ts(event.event_id)
 
                         synapse.metrics.event_processing_lag_by_event.labels(
                             "federation_sender"
                         ).observe((now - ts) / 1000)
 
-                async def handle_room_events(events: Iterable[EventBase]) -> None:
+                        return destinations
+                    return None
+
+                async def get_federatable_events_and_destinations(
+                    room_id: str, events: Iterable[EventBase]
+                ) -> Tuple[str, List[Tuple[EventBase, Collection[str]]]]:
                     with Measure(self.clock, "handle_room_events"):
```

Comment:
this Measure block needs renaming. Or, more likely, getting rid of.

Reply:
It still *does* mark some timing on how long it takes to get information for room events' destinations, so i'll rename it to `fetch_destinations_for_events`

Reply:
(note that I only see the original version of a comment in my notifications, so I didn't see the "is that alright" you added later)